src/app/features/charts/components/median-barchart/median-barchart.component.ts

```diff
@@ -0,0 +1,260 @@
+      (changes.yAxisLabel && !changes.yAxisLabel.firstChange)
+    ) {
+      if (this._data.length === 0) {
+        this.clearChart();
+        this.hideChart();
+      } else {
+        this.clearChart();
+        this.showChart();
+        this.createChart();
+      }
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (this._data.length === 0) this.hideChart();
+    else this.createChart();
+  }
+
+  ngOnDestroy(): void {
+    this.destroyChart();
+  }
+
+  clearChart() {
+    const svg = d3.select(this.chartRef.nativeElement);
+    svg.selectAll('*').remove();
+  }
+
+  hideChart() {
+    const svg = d3.select(this.chartRef.nativeElement);
+    svg.style('display', 'none');
+  }
+
+  showChart() {
+    const svg = d3.select(this.chartRef.nativeElement);
+    svg.style('display', 'block');
+  }
+
+  destroyChart() {
+    if (this.chartInitialized) this.chart.remove();
+    if (this.tooltipInitialized) this.tooltip.remove();
+  }
+
+  showTooltip(text: string, x: number, y: number): void {
+    this.tooltip = d3
+      .select(this.tooltipRef.nativeElement)
+      .style('left', `${x}px`)
+      .style('top', `${y}px`)
+      .style('display', 'block')
+      .html(text);
+    this.tooltipInitialized = true;
+  }
+
+  hideTooltip() {
+    if (this.tooltipInitialized) {
+      this.tooltip.style('display', 'none');
+    }
+  }
+
+  getBarCenterX(tissue: string, xScale: d3.ScaleBand<string>): number {
+    return (xScale(tissue) || 0) + xScale.bandwidth() / 2;
+  }
+
+  // get the current width allotted to this chart or default
+  getChartBoundingWidth(): number {
+    return (
+      d3.select(this.chartRef.nativeElement).node()?.getBoundingClientRect()
+        .width || 500
+    );
+  }
+
+  createChart() {
+    if (this._data.length > 0) {
+      const barColor = this.helperService.getColor('secondary');
+      const width = this.getChartBoundingWidth();
+      const height = 350;
+      const margin = { top: 20, right: 20, bottom: 65, left: 65 };
+      const innerWidth = width - margin.left - margin.right;
+      const innerHeight = height - margin.top - margin.bottom;
+
+      this.chart = d3
+        .select(this.chartRef.nativeElement)
+        .attr('width', width)
+        .attr('height', height)
+        .append('g')
+        .attr('transform', `translate(${margin.left}, ${margin.top})`);
+
+      // SCALES
+      const xScale = d3
+        .scaleBand()
+        .domain(this._data.map((d) => d.tissue))
+        .range([0, innerWidth])
+        .padding(0.2);
+
+      const yScale = d3
+        .scaleLinear()
+        .domain([0, this.maxValueY])
+        .nice()
+        .range([innerHeight, 0]);
+
+      // BARS
+      this.chart
+        .selectAll('.medianbars')
+        .data(this._data)
+        .enter()
+        .append('rect')
+        .attr('class', 'medianbars')
+        .attr('x', (d) => xScale(d.tissue) as number)
+        .attr('y', (d) => yScale(d.medianlogcpm || 0))
```

Comment:
Since the `medianlogcpm` field can be null, I needed to add ` || 0` (instead of `d.medianlogcpm!` which results in a lint warning: `Forbidden non-null assertion`). However, `medianlogcpm` shouldn't ever be null here, since we remove objects with null `medianlogcpm` values when setting _data. I handled the warning in a similar way throughout this component, but there's probably a better approach, so happy to adjust!

Reply:
Just as a sanity check, we might want to ask @JessterB if the data might ever be null in MongoDB and if that would possibly have any meaning in the charts.

Reply:
There are no null values in the data today. If there were nulls, I believe that we'd want to not plot them. That said, we have no plans to replace or update this data at this time. If we do replace the data someday, we should be able to remove any nulls during data processing anyway, so I wouldn't worry about it.